src/creature.cpp

```diff
@@ -179,11 +179,24 @@ bool Creature::sees( const Creature &critter ) const
                     abs( posz() - critter.posz() ) <= 1 ) ) ) {
         return false;
     }
+    if( critter.is_player() && g->u.move_mode == "crouch" ) {
```

Comment:
In this particular case casting to `player` is more preferable than referencing `g->u`.

Reply:
Could you elaborate please? How can I change code? I tried to use "player" to give NPC ability to use covers too, but I don't know how to do that.

Reply:
```c++
if( const player* p = dynamic_cast<const player*>( &critter ) ) {
    // Here you can reference p as needed.
}
```

Reply:
Thanks, your reviews really help with getting used to c++. Done.